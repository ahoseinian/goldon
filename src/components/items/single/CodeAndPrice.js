import React from 'react'
import PurchaseButton from './PurchaseButton'
import Price from './Price'

const CodeAndPrice = ({item}) => item.price
  ? (
    <div>
      <span className="text-muted">
        <span>کد محصول:</span>
        <span>{item.code}</span>
      </span>
      <div className="row">
        <div className="col-sm-7 mt-1">
          <Price amount={item.price}/>
        </div>
        <div className="col-sm-5 mt-1">
          <PurchaseButton item={item}/>
        </div>
      </div>
    </div>
  )
  : null

CodeAndPrice.propTypes = {
  item: React.PropTypes.object.isRequired
}

export default CodeAndPrice
