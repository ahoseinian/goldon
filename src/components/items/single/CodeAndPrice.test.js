import React from 'react'
import {shallow} from 'enzyme'
import CodeAndPrice from './CodeAndPrice'
import toJson from 'enzyme-to-json'

describe('CodeAndPrice', () => {
  it('renders null without price', () => {
    const wrapper = shallow(<CodeAndPrice item={{}}/>)
    expect(wrapper.html()).toEqual(null)
  })

  it('renders correctly with price', () => {
    const wrapper = shallow(<CodeAndPrice item={{price: 444}}/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
