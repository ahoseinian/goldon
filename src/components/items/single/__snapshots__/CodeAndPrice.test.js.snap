exports[`CodeAndPrice renders correctly with price 1`] = `
<div>
  <span
    className="text-muted">
    <span>
      کد محصول:
    </span>
    <span />
  </span>
  <div
    className="row">
    <div
      className="col-sm-7 mt-1">
      <Price
        amount={444} />
    </div>
    <div
      className="col-sm-5 mt-1">
      <Connect(PurchaseButton)
        item={
          Object {
            "price": 444,
          }
        } />
    </div>
  </div>
</div>
`;
